lib/distrobaker.py

```diff
@@ -766,16 +794,25 @@ def get_buildsys(which):
     if 'main' not in c:
         logger.critical('DistroBaker is not configured, aborting.')
         return None
-    if which != 'source' and which != 'destination':
+    if which not in ('source', 'destination'):
         logger.error('Cannot get "%s" build system.', which)
         return None
-    if not hasattr(get_buildsys, which):
+
+    session_timed_out = False
+    if hasattr(get_buildsys, which):
+        session_age = datetime.datetime.now() - getattr(get_buildsys, which)['session_start_time']
+        #slightly less than an hour, to be safe
+        if session_age.seconds > 3550 or session_age.days > 0:
+            session_timed_out = True
+
+    if not hasattr(get_buildsys, which) or session_timed_out:
```

Comment:
would be more readable 
```suggestion
    if session_timed_out or not hasattr(get_buildsys, which) :
```

Reply:
Changed.